app-cpp/app355.cpp

```diff
@@ -259,149 +71,64 @@ void SatOrbit::OnCalculateOrbitalDataMulti(const tle_const_iterator& tleBegin, c
-            
-            auto slice = std::make_tuple(orbitBegin, orbitEnd );
-            //IteratorPairVector::value_type slice{ orbitBegin, orbitEnd };
-            subListVector.push_back(std::move(slice));
-        }
-        
-        // Join the threads together
-        // wait for the threads to finish before returning
-        std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
-        {
-            inThread.join();
-        });
+        std::cout << "Failed to open the file " << filePath << std::endl;
+        const auto err = std::make_error_code(std::errc::io_error);
+        throw std::filesystem::filesystem_error("File could not be opened", err);
     }
 
-    // Merge the sorted sublists
-    // Use multithreading by merging multilple sublists at a time
-    // Join the threads before merging the set of larger sublists
-    // Loop until there is only one list of completely sorted OrbitalData
-    while (subListVector.size() > 1)
-    {
-        // Merge sort the sublists 2 at a time
-        std::vector<std::thread> threadVector{};
-        IteratorPairVector newSubListVector{};
+    std::vector<OrbitalData> orbitalVector;
+    std::string readLine;
 
-        // This loop will merge the sublists 2 at a time, therefore we start at 1 and increment by 2
-        for (std::size_t i = 1; i < subListVector.size(); i += 2)
-        {
-            // Get first 2 sublists to merge
-            auto& [begin, mid] = subListVector[i-1];
-            auto& [mid2, end] = subListVector[i];
-            // Mids must be sequential in memory
-            assert(mid == mid2);
-
-            // Start a thread to merge the 2 sublists
-            threadVector.push_back(std::thread(&SatOrbit::OnSortMergeVector, this, begin, mid, end));
-            // Create a new sublist element from the 2 merged sublists onto a new list
-            newSubListVector.push_back(std::make_tuple(begin, end));
-        }
-        // Wait for the threads to finish their merge operation
-        std::for_each(threadVector.begin(), threadVector.end(), [](std::thread& inThread)
-        {
-            inThread.join();
-        });
+    std::string name{};
+    std::string line1{};
+    std::string line2{};
+    
+    std::vector<sat355::TLE> tleVector;
 
-        // If the number of sublists is odd, it would be missed in the loop above; add it to the new list
-        const bool isOdd = ((subListVector.size() & 1) == 1);
-        if (isOdd)
-        {
-            newSubListVector.push_back(subListVector.back());
-        }
+    while(!fileStream.eof())
+    {
+        std::getline(fileStream, name);
+        std::getline(fileStream, line1);
+        std::getline(fileStream, line2);
 
-        // Update subListVector with the new merged results
-        subListVector = std::move(newSubListVector);
+       sat355::TLE newTLE{name, line1, line2};
+        tleVector.push_back(std::move(newTLE));
```

Comment:
good use of `std::move` 👍 
notice how you didn't declare your variable as `const`, so you can later `std::move` it 👍 👍